test/unit/types/basic_schema.test-d.ts

```diff
@@ -0,0 +1,34 @@
+import { expectType } from 'tsd';
```

Comment:
All files in this directory have to end with `test-d.ts` to be discovered by `tsd`. None of the code is actually run, its run through the TS compiler and the `expectType` etc functions allow tsd to capture the results. I tried to come up with a variety of schemas and I tested mostly with insertOne since it runs the schema through the more complex `EnhancedOmit`, `OptionalId`, and `InferIdType` types defined in mongo_types.ts

Please let me know if there's a case y'all think we should add here. 🙂